src/js/angular/core/services/jwt-auth.service.js

```diff
@@ -3,294 +3,302 @@ import 'angular/core/services/openid-auth.service.js';
 import 'angular/rest/security.rest.service';
 import {UserRole} from 'angular/utils/user-utils';
 
+const LOGIN_PATH = '/login';
+const NO_ACCESS = 'noaccess';
+const EXPIRED = 'expired';
+const TOKEN_TYPE_BEARER = 'Bearer';
+const OVERRIDE_AUTH_USERNAME = 'overrideauth';
+const ADMIN_USERNAME = 'admin';
+const READ_PERMISSION = 'READ';
+const WRITE_PERMISSION = 'WRITE';
+const TOKEN_TYPE_GDB = 'GDB';
+const AUTH_FULLY = 'IS_AUTHENTICATED_FULLY';
+const ROLE_SEPARATOR = '_';
+const SYSTEM_REPO = 'SYSTEM';
+const REPO_WILDCARD = '*';
+
 angular.module('graphdb.framework.core.services.jwtauth', [
     'toastr',
     'graphdb.framework.rest.security.service',
     'graphdb.framework.core.services.openIDService'
 ])
     .service('$jwtAuth', ['$http', 'toastr', '$location', '$rootScope', 'SecurityRestService', '$openIDAuth', '$translate', '$q', 'AuthTokenService', 'LSKeys', 'LocalStorageAdapter',
         function ($http, toastr, $location, $rootScope, SecurityRestService, $openIDAuth, $translate, $q, AuthTokenService, LSKeys, LocalStorageAdapter) {
-            const jwtAuth = this;
 
+            // =========================
+            // Public variables
+            // =========================
             $rootScope.deniedPermissions = {};
-            $rootScope.setPermissionDenied = function (path) {
-                if (path === '/login' || !jwtAuth.isAuthenticated()) {
+
+            // =========================
+            // Private variables
+            // =========================
+            let _securityEnabled = true;
+            let _externalAuthUser = false;
+            let _principal = undefined;
+            let _freeAccessPrincipal = undefined;
+            let _freeAccess = false;
+            let _securityInitialized = false;
+            let _hasOverrideAuth = false;
+            let _externalAuth = undefined;
+            let _authImplementation = undefined;
+            let _openIDEnabled = false;
+            let _passwordLoginEnabled = undefined;
+            let _openIDConfig = undefined;
+            let _gdbUrl = undefined;
+
+            // =========================
+            // RootScope functions
```

Comment:
All these rootScope shits usually should be just public functions exposed by the service and used wherever needed in controllers. I'm not asking you to do do this, but only pointing out a really bad case of tight coupling .